Sources/SKOptions/PreparationBatchingStrategy.swift

```diff
@@ -0,0 +1,54 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Defines the batch size for target preparation.
+///
+/// If nil, SourceKit-LSP will default to preparing 1 target at a time.
+///
+/// - discriminator: strategy
+public enum PreparationBatchingStrategy: Sendable, Equatable {
+  /// Prepare a fixed number of targets in a single batch.
+  ///
+  /// `batchSize`: The number of targets to prepare in each batch.
+  case target(batchSize: Int)
```

Comment:
`target` is pretty ambiguous here, should we make this more specific and call it `fixedTargetBatchSize`. That should really clarify what this strategy does.